Xamarin.Forms.Core/Button.cs

```diff
@@ -50,6 +50,29 @@ public class Button : View, IFontElement, ITextElement, IBorderElement, IButtonC
 			propertyChanging: (bindable, oldvalue, newvalue) => ((Button)bindable).OnSourcePropertyChanging((ImageSource)oldvalue, (ImageSource)newvalue),
 			propertyChanged: (bindable, oldvalue, newvalue) => ((Button)bindable).OnSourcePropertyChanged((ImageSource)oldvalue, (ImageSource)newvalue));
 
+		public static readonly BindableProperty PaddingProperty = PaddingElement.PaddingProperty;
+
+		public Thickness Padding
+		{
+			get { return (Thickness)GetValue (PaddingElement.PaddingProperty); }
+			set { SetValue (PaddingElement.PaddingProperty, value); }
+		}
+
+		Thickness IPaddingElement.PaddingDefaultValueCreator ()
+		{
+			return default (Thickness);
+		}
+
+		void IPaddingElement.OnPaddingPropertyChanged (Thickness oldValue, Thickness newValue)
+		{
+			UpdatePadding (newValue);
```

Comment:
Invalidate Measure
When the padding is changed the measure on the button is going to change so that needs to Invalidate the Measure on the button.  You can see the issue with this if you set the padding after the button has renderered vs before.  If you don't invalidate the measure then it would look different in each case because the height/width won't adjust to the new padding